src/main/java/com/minecolonies/commands/DeleteColonyCommand.java

```diff
@@ -0,0 +1,122 @@
+package com.minecolonies.commands;
+
+import com.minecolonies.colony.ColonyManager;
+import com.minecolonies.colony.IColony;
+import com.mojang.authlib.GameProfile;
+import net.minecraft.command.CommandException;
+import net.minecraft.command.ICommandSender;
+import net.minecraft.server.MinecraftServer;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.text.TextComponentString;
+import org.jetbrains.annotations.NotNull;
+
+import javax.annotation.Nullable;
+import java.util.Collections;
+import java.util.List;
+import java.util.UUID;
+
+/**
+ * List all colonies.
+ */
+public class DeleteColonyCommand extends AbstractSingleCommand
+{
+
+    public static final  String DESC                       = "delete";
+    private static final String NO_COLONY_FOUND_MESSAGE    = "Colony with mayor %s not found.";
+    private static final String NO_COLONY_FOUND_MESSAGE_ID = "Colony with ID %d not found.";
+
+    /**
+     * Initialize this SubCommand with it's parents.
+     *
+     * @param parents an array of all the parents.
+     */
+    public DeleteColonyCommand(@NotNull final String... parents)
+    {
+        super(parents);
+    }
+
+    @NotNull
+    @Override
+    public String getCommandUsage(@NotNull final ICommandSender sender)
+    {
+        return super.getCommandUsage(sender) + "<ColonyId|OwnerName>";
+    }
+
+    @Override
+    public void execute(@NotNull final MinecraftServer server, @NotNull final ICommandSender sender, @NotNull final String... args) throws CommandException
+    {
+        int colonyId = -1;
+        UUID mayorID = null;
+
+        if (args.length != 0)
+        {
+            try
+            {
+                colonyId = Integer.parseInt(args[0]);
+            }
+            catch (NumberFormatException e)
+            {
+                mayorID = getUUIDFromName(sender, args);
+            }
+        }
+
```

Comment:
It's an edge case, but if someone's name is only numbers we could run into an issue.